crypto/README.md

```diff
@@ -49,3 +84,31 @@ Produces a signature of the given message using the private key.
 **`verifySignature = (message: Plaintext, signature: Signature, publicKey: PubKey): boolean`**
 
 Checks whether the given signature is valid.
+
+### Hash functions
+
+**`sha256Hash = (input: BigInt[]): BigInt`**: a wrapper function over
+`ethers.utils.soliditySha256`, where the output is modulo the BN254 group
+order.
+
+**`hashOne = (elements: Plaintext): BigInt`**: the Poseidon hash function for
+one input. Equivalent to `hash2([input, 0])`.
+
+**`hash2 = (elements: Plaintext): BigInt`**: the Poseidon hash function for 2
+inputs.
+
+**`hashLeftRight = (left: BigInt, right: BigInt): BigInt`**: equivalent to
+`hash2([left, right])`.
+
+**`hash3 = (elements: Plaintext): BigInt`**: the Poseidon hash function for 3
+inputs.
+
+**`hash4 = (elements: Plaintext): BigInt`**: the Poseidon hash function for 4
+inputs.
+
+**`hash5 = (elements: Plaintext): BigInt`**: the Poseidon hash function for 5
+inputs.
+
+**`hash12 = (elements: Plaintext): BigInt`**: the Poseidon hash function for 12
```

Comment:
Does this implicitly use hashLeftRight by first using hash5 then hash6 on the remaining inputs? If so we may want to include this here so users are aware of the ordering.

Reply:
Thanks for the suggestion. I've added the pseudocode in a new commit.